src/metabase/api/public.clj

```diff
@@ -614,10 +614,14 @@
      :qp            qp.pivot/run-pivot-query)))
 
 (def ^:private action-execution-throttle
-  "Rate limit at 1 action per second on a per action basis.
+  "Rate limit at 1 action per 100 ms (10 per second) on a per action basis.
    The goal of rate limiting should be to prevent very obvious abuse, but it should
    be relatively lax so we don't annoy legitimate users."
-  (throttle/make-throttler :action-uuid :attempts-threshold 1 :initial-delay-ms 1000 :delay-exponent 1))
+  (throttle/make-throttler :action-uuid
+                           :attempts-threshold 1
+                           :initial-delay-ms 100
+                           :attempt-ttl-ms 100
+                           :delay-exponent 1))
```

Comment:
I may not be understanding how this works correctly, but I think you just want to increase the `attempt-threshhold`
```suggestion
  (throttle/make-throttler :action-uuid
                           :attempts-threshold 10
                           :initial-delay-ms 1000
                           :attempt-ttl-ms 1000
                           :delay-exponent 1))
```
turns out this library is pretty well documented! 😂 
https://github.com/metabase/throttle

Reply:
You're right, the way I've implemented it is probably little too strict, and has a higher chance of blocking legitimate users by chance if there are two requests in a given 100 ms. 

I was a little unsure about your suggestion because the throttle library actually seems to have a bug in it. Here is a demonstration of some behaviour under your suggestion's settings:

```
attempt : time : status
1 0.0 succeed
2 0.1 succeed
3 0.2 succeed
4 0.3 succeed
5 0.4 succeed
6 0.5 succeed
7 0.6 succeed
8 0.7 succeed
9 0.8 succeed
10 0.9 succeed
11a 0.99 fail <------ this one will fail because it is the 10 successful attempts in the last second
11b 0.99 fail <------ this one will fail because there are 10 successful attempts in the last second
11c 0.99 fail <------ this one will fail because it is the 11th in the last second
11d 0.99 fail <------ this one will fail because it is the 11th in the last second
12 1.0 succeed <--- this one will succeed, because there have been 9 successful attempts in the last second
13 1.1 succeed
14 1.2 succeed
15 1.3 succeed
16 1.4 succeed
17 1.5 succeed
18 1.6 succeed
19 1.7 succeed
20 1.8 succeed
... etc

```

I consider this a bug because it doesn't matter what `delay-exponent` or `initial-delay-ms` is, the behaviour is the same. But practically this bug doesn't matter much for the use case. If we just want to rate limit to 10 requests per second then this works too.

Reply:
well, if you want I can introduce you to the throttle library's maintainer 😂

Reply:
For my future reference, here is where we explicitly don't count failed attempts towards the attempt count:
https://github.com/metabase/throttle/blob/master/src/throttle/core.clj#L64

And here is where we remove successful attempts past the TTL
https://github.com/metabase/throttle/blob/master/src/throttle/core.clj#L119